cardano-db-sync/src/Cardano/DbSync/Sync.hs

```diff
@@ -234,7 +234,7 @@ dbSyncProtocols syncEnv metricsSetters tc _version codecs _connectionId =
                     ( Client.chainSyncClientPeer $
                         chainSyncClientFixScripts backend tracer ls
                     )
-              when onlyFix $ panic "All Good! This error is only thrown to exit db-sync" -- TODO fix.
+              when onlyFix $ error "All Good! This error is only thrown to exit db-sync"
```

Comment:
I know you didn't introduce this, but I think it would be better to find a way to exit gracefully here.

Reply:
I think it's impossible to exit gratefully and this is a limitation of either the protocol or the network-client library that we use. 
This is one of the cases where `panic` should remain. Otherwise the connection will keep restarting.